api/client/commands.go

```diff
@@ -162,21 +163,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 			}
 
 			// Verify that 'filename' is within the build context
-			if !strings.HasSuffix(absRoot, string(os.PathSeparator)) {
-				absRoot += string(os.PathSeparator)
-			}
-			if !strings.HasPrefix(filename, absRoot) {
-				return fmt.Errorf("The Dockerfile (%s) must be within the build context (%s)", *dockerfileName, root)
+			filename, err = symlink.FollowSymlinkInScope(filename, absRoot)
```

Comment:
IMO these two should act the same - if they don't then I think its a bug that we need to fix.
docker build -f Dockerfile . == docker build .